### PR TITLE
Improve mlos-viz for multiple repeats of a config and add tests

### PR DESCRIPTION
**Pull request by bpkroth**
- Mark `mlos_viz` as `typed` for `mypy`
- Bump version
- Mock calls to matplotlib/dabl for testing
- Add plotting of top-N configs
- Improve plots for handling repeat config trials via variance error bars

### CODE REVIEW

File: mlos_bench/mlos_bench/launcher.py [hunk 1]
```diff
@@ -340,6 +340,10 @@ def _load_optimizer(self, args_optimizer: Optional[str]) -> Optimizer:
             return OneShotOptimizer(
                 self.tunables, config=config, service=self._parent_service)
         class_config = self._config_loader.load_config(args_optimizer, ConfigSchema.OPTIMIZER)
+        if 'max_iterations' in class_config:
+            class_config['max_iterations'] *= self.trial_config_repeat_count
+        if 'max_iterations' in self.global_config:
+            self.global_config['max_iterations'] *= self.trial_config_repeat_count
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/launcher.py:346]: - [ ] separate PR with tests

File: mlos_bench/mlos_bench/storage/base_experiment_data.py [hunk 2]
```diff
@@ -134,3 +134,94 @@ def results_df(self) -> pandas.DataFrame:
...
             trial was not successful.
         """
+
+    def top_n_configs(self,
+                      *,
+                      top_n_configs: int = 20,
+                      objective_name: Optional[str] = None,
+                      method: Union[Literal["mean", "median"], float] = "mean",
+                      ) -> Tuple[pandas.DataFrame, str, str]:
+        # pylint: disable=too-complex
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/storage/base_experiment_data.py:144]: - [ ] simplify with new `TunableConfigTrialGroup`s

File: mlos_bench/mlos_bench/launcher.py (hunk 1 above)
Comment (bpkroth) [reply mlos_bench/mlos_bench/launcher.py:346]: #651

File: mlos_bench/mlos_bench/storage/base_experiment_data.py [hunk 3]
```diff
@@ -119,6 +120,33 @@ def tunable_config_trial_groups(self) -> Dict[int, "TunableConfigTrialGroupData"
             A dictionary of the trials' data, keyed by (tunable) by config id.
         """
 
+    @property
+    def default_tunable_config_id(self) -> Optional[int]:
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/storage/base_experiment_data.py:124]: Split out to #650

**Review by motus — APPROVED** [review]
Looks good!

**Pull request merged**